x-pack/metricbeat/module/iis/application_pool/reader.go

```diff
@@ -62,87 +63,138 @@ var appPoolCounters = map[string]string{
 }
 
 // newReader creates a new instance of Reader.
-func newReader() (*Reader, error) {
+func newReader(config Config) (*Reader, error) {
 	var query pdh.Query
 	if err := query.Open(); err != nil {
 		return nil, err
 	}
-	reader := &Reader{
-		Query: query,
-		log:   logp.NewLogger("website"),
+	r := &Reader{
+		query:           query,
+		log:             logp.NewLogger("application_pool"),
+		config:          config,
+		workerProcesses: make(map[string]string),
+	}
+	apps, err := getApplicationPools(config.Names)
+	if err != nil {
+		return r, errors.Wrap(err, "failed retrieving running worker processes")
```

Comment:
If something fails here, should we close `query` to avoid leaks?

Reply:
indeed, added the call